File: testdata/jennies/builders/map_index_assignment/PHPBuilder/src/Sandbox/SomeStructBuilder.php
<?php

namespace Grafana\Foundation\Sandbox;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Sandbox\SomeStruct>
 */
class SomeStructBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\Sandbox\SomeStruct $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\Sandbox\SomeStruct();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\Sandbox\SomeStruct
     */
    public function build()
    {
        return $this->internal;
    }

    public function annotations(string $key, string $value): static
    {
        $this->internal->annotations[$key] = $value;
    
        return $this;
    }

}

File: testdata/jennies/builders/map_index_assignment/CSharpBuilder/src/Grafana/Foundation/Sandbox/SomeStructBuilder.cs
namespace Grafana.Foundation.Sandbox;


public class SomeStructBuilder : Cog.IBuilder<SomeStruct>
{
    protected readonly SomeStruct @internal;

    public SomeStructBuilder()
    {
        this.@internal = new SomeStruct();
    }

    public SomeStructBuilder Annotations(string key,string value)
    {
        if (this.@internal.Annotations == null)
        {
            this.@internal.Annotations = new Dictionary<string, string>();
        }
        this.@internal.Annotations[key] = value;
        return this;
    }

    public SomeStruct Build()
    {
        return this.@internal;
    }
}
